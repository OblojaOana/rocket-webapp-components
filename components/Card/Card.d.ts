export interface CardProps {
    /**
    * The className of card.
    */
    className?: String;
    /**
     * The color of card.
     */
    color?: "primary" |
            "info" |
            "success" |
            "warning" |
            "danger" |
            "rose"
    /**
    * The content of card.
    */
    children?: React.ReactNode
}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/card
 *
 * 
 */
export default function Card(props: CardProps): JSX.Element;