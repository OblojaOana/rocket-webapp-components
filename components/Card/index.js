import Card from "../../core/Card";
export default Card;