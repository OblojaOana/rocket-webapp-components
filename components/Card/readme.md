##### Copy this snapshot in the overview panel
 ```js
import React from 'react';
import Card from '@bit/totalsoft_oss.react-mui.card';

export default (
	<Card>
		<div>Card</div>
	</Card>
)
 ```