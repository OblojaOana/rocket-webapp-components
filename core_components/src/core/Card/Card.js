import React from "react";
import classNames from "classnames";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core/styles";
import cardStyle from "./cardStyle";

const useStyles = makeStyles(cardStyle);

export default function Card({ className, color, children, ...rest }) {
  const classes = useStyles();
  const cardClasses = classNames({
    [classes.card]: true,
    [classes[color]]: color,
    [className]: className !== undefined
  });

  return (
    <div className={cardClasses} {...rest}>
      {children}
    </div>
  );
}

Card.propTypes = {
  className: PropTypes.string,
  color: PropTypes.oneOf([
    "primary",
    "info",
    "success",
    "warning",
    "danger",
    "rose"
  ]),
  children: PropTypes.node
};