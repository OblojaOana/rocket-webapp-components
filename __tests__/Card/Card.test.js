import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../../components/Card/Card';

describe('Card', () => {
    it('snapshot - primary', () => {
        const component = renderer.create(<Card cololr="primary">
            <div> Card</ div>
        </Card>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('snapshot - info', () => {
        const component = renderer.create(<Card cololr="info">
            <div> Card</ div>
        </Card>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('snapshot - success', () => {
        const component = renderer.create(<Card cololr="sucess">
            <div> Card</ div>
        </Card>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('snapshot - warning', () => {
        const component = renderer.create(<Card cololr="warning">
            <div> Card</ div>
        </Card>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('snapshot - danger', () => {
        const component = renderer.create(<Card cololr="danger">
            <div> Card</ div>
        </Card>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('snapshot - rose', () => {
        const component = renderer.create(<Card cololr="rose">
            <div> Card</ div>
        </Card>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});