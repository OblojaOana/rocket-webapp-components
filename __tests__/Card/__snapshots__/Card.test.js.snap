// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot - danger 1`] = `
<div
  className="makeStyles-card-1"
>
  <div>
     Card
  </div>
</div>
`;

exports[`Card snapshot - info 1`] = `
<div
  className="makeStyles-card-1"
>
  <div>
     Card
  </div>
</div>
`;

exports[`Card snapshot - primary 1`] = `
<div
  className="makeStyles-card-1"
>
  <div>
     Card
  </div>
</div>
`;

exports[`Card snapshot - rose 1`] = `
<div
  className="makeStyles-card-1"
>
  <div>
     Card
  </div>
</div>
`;

exports[`Card snapshot - success 1`] = `
<div
  className="makeStyles-card-1"
>
  <div>
     Card
  </div>
</div>
`;

exports[`Card snapshot - warning 1`] = `
<div
  className="makeStyles-card-1"
>
  <div>
     Card
  </div>
</div>
`;
